sdk/loadtestservice/load-testing-rest/src/getFileValidationPoller.ts

```diff
@@ -0,0 +1,165 @@
+  );
+  type Handler = (state: OperationState<TestGetFile200Response>) => void;
+
+  const state: OperationState<TestGetFile200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestGetFile200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestGetFile200Response>, TestGetFile200Response> = {
+    async poll(options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      if (options?.abortSignal?.aborted) {
+        state.error = new Error("The operation was aborted.");
+        return;
+      }
+
+      if (fileName) {
+        const fileValidationResponse = await client
+          .path("/tests/{testId}/files/{fileName}", testId, fileName)
+          .get();
+        if (isUnexpected(fileValidationResponse)) {
+          state.status = "failed";
+          state.error = new Error(fileValidationResponse.body.error.message);
+          return;
+        }
+
+        switch (fileValidationResponse.body.validationStatus) {
+          case "NOT_VALIDATED": {
+            if (fileValidationResponse.body.fileType === "JMX_FILE") {
+              state.status = "running";
+            } else {
+              state.status = "succeeded";
+            }
+            break;
+          }
+          case "VALIDATION_INITIATED": {
+            state.status = "running";
+            break;
+          }
+          case "VALIDATION_SUCCESS":
+          case "VALIDATION_NOT_REQUIRED": {
+            state.status = "succeeded";
+            break;
+          }
+          case "VALIDATION_FAILURE": {
+            state.status = "failed";
+            state.error = new Error(fileValidationResponse.body.validationStatus);
+            break;
+          }
+        }
+        state.result = fileValidationResponse;
+
+        await processProgressCallbacks();
+      }
+    },
+
+    pollUntilDone(pollOptions?: {
+      abortSignal?: AbortSignalLike;
+    }): Promise<TestGetFile200Response> {
+      return (resultPromise ??= (async () => {
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        const { signal: abortSignal } = inputAbortSignal
+          ? new AbortController([inputAbortSignal, abortController.signal])
+          : abortController;
+        if (!poller.isDone()) {
+          await poller.poll({ abortSignal });
+          while (!poller.isDone()) {
+            const delay = sleep(currentPollIntervalInMs, abortSignal);
+            cancelJob = () => abortController.abort();
+            try {
+              await delay;
+            } catch (ex: any) {
+              state.status = "canceled";
+            }
```

Comment:
The operation can't be determined to be canceled here.
```suggestion
            await delay;
```

Reply:
After offline discussion, we can swallow the error and just break;
```suggestion
            try {
              await delay;
            } catch (ex) {
              break;
            }
```

Reply:
Updated with error thrown when timed out, as discussed offline.